pkg/reconciler/service/occupancytracker.go

```diff
@@ -0,0 +1,125 @@
+package service
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"github.com/kyma-incubator/reconciler/pkg/logger"
+	"github.com/kyma-incubator/reconciler/pkg/reconciler"
+	"go.uber.org/zap"
+	"net/http"
+	"net/url"
+	"os"
+	"time"
+)
+
+const occupancyURLTemplate = "%s://%s:%s/v1/occupancy/%s"
+
+type OccupancyTracker struct {
+	logger               *zap.SugaredLogger
+	occupancyID          string
+	occupancyCallbackURL string
+}
+
+func newOccupancyTracker(debug bool) *OccupancyTracker {
+	return &OccupancyTracker{
+		logger: logger.NewLogger(debug),
+	}
+}
+
+func (t *OccupancyTracker) Track(ctx context.Context, pool *WorkerPool, reconcilerName string) {
+	podName, err := os.Hostname()
+	if err != nil {
+		t.logger.Errorf("occupancy tracker could not retrieve pod name: %s", err)
+		return
+	}
+	//using hostname (= pod name) as the id to be able
+	//to clean up pods that have crashed w/o being able to delete their occupancy
+	t.occupancyID = podName
+	go func() {
+		ticker := time.NewTicker(defaultInterval)
+		for {
+			select {
+			case <-ctx.Done():
+				if t.occupancyCallbackURL != "" && !pool.IsClosed() {
+					t.logger.Info("occupancy tracker is deleting Worker Pool occupancy")
+					t.deleteWorkerPoolOccupancy()
+					return
+				}
+
+			case <-ticker.C:
+				if t.occupancyCallbackURL != "" && !pool.IsClosed() {
+					t.createOrUpdateComponentReconcilerOccupancy(reconcilerName, pool.RunningWorkers(), pool.Size())
+				}
+			}
+		}
+
+	}()
+}
+
+func (t *OccupancyTracker) createOrUpdateComponentReconcilerOccupancy(reconcilerName string, runningWorkers, poolSize int) {
+	httpOccupancyUpdateRequest := reconciler.HTTPOccupancyRequest{
+		Component:      reconcilerName,
+		RunningWorkers: runningWorkers,
+		PoolSize:       poolSize,
+	}
+	jsonPayload, err := json.Marshal(httpOccupancyUpdateRequest)
+	if err != nil {
+		t.logger.Errorf("occupancy tracker failed to marshal HTTP payload to update occupancy of service '%s': %s", t.occupancyID, err)
+		return
+	}
+	resp, err := http.Post(t.occupancyCallbackURL, "application/json", bytes.NewBuffer(jsonPayload))
+	if err != nil {
+		t.logger.Error(err.Error())
+		return
+	}
+
+	if resp.StatusCode < http.StatusOK || resp.StatusCode > 299 {
+		t.logger.Errorf("mothership failed to update occupancy for '%s' service with status code: '%d'", t.occupancyID, resp.StatusCode)
```

Comment:
Would treat it as warning - error means normally that a serious issue happend. This situation can also happen when the mothership is temporarily down.

Reply:
Thanks! Will modify it in the subsequent PR 👍